app/views/Home/index.tsx

```diff
@@ -183,13 +121,49 @@ function Home(props: ViewProps) {
         },
     );
 
+    const {
+        data: pinnedProjectsResponse,
+        loading: pinnedProjectsPending,
+        refetch: retriggerPinnedProjectsList,
+    } = useQuery<UserPinnedProjectsQuery, UserPinnedProjectsQueryVariables>(
+        USER_PINNED_PROJECTS,
+        {
+            onCompleted: (response) => {
+                const count = response?.userPinnedProjects.length;
+                setPinButtonDisabled(count >= MAX_PINNED_PROJECT_LIMIT);
```

Comment:
instead of setting it in a state variable, read it directly from the data and disable/enable it.